alerting/src/main/kotlin/org/opensearch/alerting/MonitorRunnerService.kt

```diff
@@ -183,28 +196,45 @@ object MonitorRunnerService : JobRunner, CoroutineScope, AbstractLifecycleCompon
         runnerSupervisor.cancel()
     }
 
-    override fun doClose() { }
+    override fun doClose() {}
 
     override fun postIndex(job: ScheduledJob) {
-        if (job !is Monitor) {
+        if (job is Monitor) {
+            launch {
+                try {
+                    monitorCtx.moveAlertsRetryPolicy!!.retry(logger) {
+                        if (monitorCtx.alertIndices!!.isAlertInitialized(job.dataSources)) {
+                            moveAlerts(monitorCtx.client!!, job.id, job)
+                        }
+                    }
+                } catch (e: Exception) {
+                    logger.error("Failed to move active alerts for monitor [${job.id}].", e)
+                }
+            }
+        } else if (job is Workflow) {
+            launch {
+                try {
+                    monitorCtx.moveAlertsRetryPolicy!!.retry(logger) {
+                        moveAlerts(monitorCtx.client!!, job.id, job, monitorCtx)
```

Comment:
We need to check if the alerts index is initialized if its not using a different datasource. Move alerts checks only if the datasource alerts index is initialized if there is a data source.

Reply:
datasource is always there
for monitors created directly by alerting plugin through rest api or dashboards we populate them with defaults

Reply:
https://github.com/opensearch-project/common-utils/blob/main/src/main/kotlin/org/opensearch/commons/alerting/model/DataSources.kt#L13-L39